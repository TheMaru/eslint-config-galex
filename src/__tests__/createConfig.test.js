const fs = require('fs');
const { join } = require('path');
const readPkgUp = require('read-pkg-up');

const { createConfig, getDependencies } = require('../createConfig');

describe('getDependencies', () => {
  test('matches snapshot', () => {
    expect(getDependencies()).toMatchSnapshot();
  });

  test('matches snapshot when erroring', () => {
    // eslint-disable-next-line no-console
    const consoleError = console.error;

    const consoleErrorSpy = jest
      .spyOn(console, 'error')
      .mockImplementation(() => {});

    expect(getDependencies({ cwd: '/foo' })).toMatchSnapshot();
    expect(consoleErrorSpy).toHaveBeenCalledTimes(1);

    // eslint-disable-next-line no-console
    console.error = consoleError;
  });

  test('accepts alternative tsdefaultConfig path', () => {
    const cwd = join(process.cwd(), 'src', '__tests__');

    expect(getDependencies({ cwd })).toMatchSnapshot();
  });
});

describe('createConfig', () => {
  test('matches snapshot', () => {
    expect(createConfig()).toMatchSnapshot();
  });

  test('given typescript, determines env.node based on @types/node', () => {
    const defaultConfig = createConfig();

    jest.spyOn(readPkgUp, 'sync').mockReturnValueOnce({
      packageJson: {
        dependencies: {
          typescript: '1.0.0',
        },
      },
    });

    jest.spyOn(fs, 'accessSync').mockReturnValueOnce(true);

    const config = createConfig();

    expect(config.env.node).toBeFalsy();
    expect(config.env.node).not.toBe(defaultConfig.env.node);
  });

  test('allows passing extra rules', () => {
    const config = createConfig({ customRules: { foo: 'bar' } });

    expect(config.rules.foo).toBe('bar');
    expect(config).toMatchSnapshot();
  });
});
