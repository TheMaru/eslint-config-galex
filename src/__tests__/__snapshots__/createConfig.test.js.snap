// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createConfig allows passing extra rules 1`] = `
Object {
  "env": Object {
    "browser": false,
    "es2020": true,
    "node": true,
  },
  "overrides": Array [
    Object {
      "env": Object {
        "jest": true,
      },
      "extends": Array [
        "plugin:jest-formatting/strict",
      ],
      "files": Array [
        "**/*.?(test|spec).?(ts|js)?(x)",
      ],
      "parserOptions": Object {
        "ecmaVersion": 2020,
      },
      "plugins": Array [
        "jest",
      ],
      "rules": Object {
        "import/no-namespace": "off",
        "jest/consistent-test-it": "off",
        "jest/expect-expect": "off",
        "jest/lowercase-name": "off",
        "jest/no-alias-methods": "off",
        "jest/no-commented-out-tests": "warn",
        "jest/no-conditional-expect": "error",
        "jest/no-deprecated-functions": "error",
        "jest/no-disabled-tests": "warn",
        "jest/no-duplicate-hooks": "warn",
        "jest/no-expect-resolves": "off",
        "jest/no-export": "error",
        "jest/no-focused-tests": "error",
        "jest/no-hooks": "off",
        "jest/no-identical-title": "error",
        "jest/no-if": "error",
        "jest/no-interpolation-in-snapshots": "error",
        "jest/no-jasmine-globals": "error",
        "jest/no-jest-import": "error",
        "jest/no-large-snapshots": Array [
          "warn",
          Object {
            "maxSize": 300,
          },
        ],
        "jest/no-mocks-import": "error",
        "jest/no-restricted-matchers": "off",
        "jest/no-standalone-expect": "off",
        "jest/no-test-callback": "error",
        "jest/no-test-prefixes": "error",
        "jest/no-test-return-statement": "error",
        "jest/no-truthy-falsy": "off",
        "jest/no-try-expect": "off",
        "jest/prefer-called-with": "error",
        "jest/prefer-expect-assertions": "off",
        "jest/prefer-hooks-on-top": "error",
        "jest/prefer-inline-snapshots": "off",
        "jest/prefer-spy-on": "warn",
        "jest/prefer-strict-equal": "warn",
        "jest/prefer-to-be-null": "warn",
        "jest/prefer-to-be-undefined": "warn",
        "jest/prefer-to-contain": "warn",
        "jest/prefer-to-have-length": "warn",
        "jest/prefer-todo": "warn",
        "jest/require-to-throw-message": "off",
        "jest/require-top-level-describe": "error",
        "jest/valid-describe": "error",
        "jest/valid-expect": "error",
        "jest/valid-expect-in-promise": "error",
        "jest/valid-title": "warn",
        "no-empty-function": "off",
      },
      "settings": Object {
        "jest": Object {
          "version": "detect",
        },
      },
    },
  ],
  "parserOptions": Object {
    "sourceType": "module",
  },
  "plugins": Array [
    "import",
    "sort-keys-fix",
    "unicorn",
    "promise",
    "sonarjs",
    "inclusive-language",
  ],
  "rules": Object {
    "array-callback-return": Array [
      "error",
      Object {
        "checkForEach": true,
      },
    ],
    "arrow-body-style": 0,
    "constructor-super": "error",
    "curly": 0,
    "default-case": "error",
    "default-case-last": "error",
    "default-param-last": "error",
    "dot-notation": "warn",
    "eqeqeq": "warn",
    "foo": "bar",
    "for-direction": "error",
    "func-names": Array [
      "warn",
      "as-needed",
    ],
    "getter-return": "warn",
    "guard-for-in": "warn",
    "import/default": "warn",
    "import/dynamic-import-chunkname": "warn",
    "import/export": "error",
    "import/first": "error",
    "import/named": "error",
    "import/namespace": "error",
    "import/newline-after-import": "warn",
    "import/no-amd": "error",
    "import/no-anonymous-default-export": "error",
    "import/no-cycle": Array [
      "warn",
      Object {
        "ignoreExternal": true,
        "maxDepth": 5,
      },
    ],
    "import/no-default-export": "warn",
    "import/no-deprecated": "warn",
    "import/no-duplicates": "warn",
    "import/no-dynamic-require": "warn",
    "import/no-extraneous-dependencies": "warn",
    "import/no-mutable-exports": "error",
    "import/no-named-default": "error",
    "import/no-namespace": "error",
    "import/no-self-import": "error",
    "import/no-unresolved": "warn",
    "import/no-unused-modules": "warn",
    "import/no-useless-path-segments": Array [
      "error",
      Object {
        "noUselessIndex": true,
      },
    ],
    "import/no-webpack-loader-syntax": "error",
    "import/order": Array [
      "warn",
      Object {
        "alphabetize": Object {
          "order": "asc",
        },
        "groups": Array [
          Array [
            "builtin",
            "external",
            "internal",
          ],
          Array [
            "unknown",
            "parent",
            "sibling",
          ],
          "index",
        ],
        "newlines-between": "always",
      },
    ],
    "inclusive-language/use-inclusive-words": "error",
    "lines-around-comment": 0,
    "lines-between-class-members": "warn",
    "max-len": 0,
    "new-cap": "warn",
    "no-alert": "error",
    "no-array-constructor": "error",
    "no-async-promise-executor": "error",
    "no-await-in-loop": "error",
    "no-bitwise": "warn",
    "no-caller": "error",
    "no-case-declarations": "warn",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": Array [
      "warn",
      "except-parens",
    ],
    "no-confusing-arrow": 0,
    "no-console": "warn",
    "no-const-assign": "error",
    "no-constant-condition": "error",
    "no-constructor-return": "error",
    "no-control-regex": "warn",
    "no-debugger": "warn",
    "no-div-regex": "warn",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-else-if": "warn",
    "no-dupe-keys": "warn",
    "no-duplicate-case": "warn",
    "no-else-return": "warn",
    "no-empty": "warn",
    "no-empty-character-class": "warn",
    "no-empty-function": "error",
    "no-empty-pattern": "warn",
    "no-eq-null": "error",
    "no-eval": "error",
    "no-ex-assign": "warn",
    "no-extend-native": "error",
    "no-extra-bind": "warn",
    "no-extra-boolean-cast": "warn",
    "no-fallthrough": "warn",
    "no-func-assign": "warn",
    "no-global-assign": "warn",
    "no-implicit-coercion": Array [
      "warn",
      Object {
        "boolean": false,
        "number": true,
        "string": true,
      },
    ],
    "no-implicit-globals": "error",
    "no-implied-eval": "error",
    "no-import-assign": "error",
    "no-inner-declarations": "warn",
    "no-invalid-regexp": "error",
    "no-invalid-this": "error",
    "no-irregular-whitespace": "warn",
    "no-labels": "error",
    "no-lonely-if": "warn",
    "no-loop-func": "error",
    "no-loss-of-precision": "error",
    "no-misleading-character-class": "warn",
    "no-mixed-operators": 0,
    "no-multi-assign": "error",
    "no-negated-condition": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-object": "error",
    "no-new-wrappers": "error",
    "no-obj-calls": "error",
    "no-octal": "warn",
    "no-octal-escape": "error",
    "no-param-reassign": "error",
    "no-promise-executor-return": "error",
    "no-proto": "error",
    "no-prototype-builtins": "error",
    "no-redeclare": "error",
    "no-regex-spaces": "warn",
    "no-restricted-globals": Array [
      "error",
      "addEventListener",
      "blur",
      "close",
      "closed",
      "confirm",
      "defaultStatus",
      "defaultstatus",
      "event",
      "external",
      "find",
      "focus",
      "frameElement",
      "frames",
      "history",
      "innerHeight",
      "innerWidth",
      "length",
      "location",
      "locationbar",
      "menubar",
      "moveBy",
      "moveTo",
      "name",
      "onblur",
      "onerror",
      "onfocus",
      "onload",
      "onresize",
      "onunload",
      "open",
      "opener",
      "opera",
      "outerHeight",
      "outerWidth",
      "pageXOffset",
      "pageYOffset",
      "parent",
      "print",
      "removeEventListener",
      "resizeBy",
      "resizeTo",
      "screen",
      "screenLeft",
      "screenTop",
      "screenX",
      "screenY",
      "scroll",
      "scrollbars",
      "scrollBy",
      "scrollTo",
      "scrollX",
      "scrollY",
      "self",
      "status",
      "statusbar",
      "stop",
      "toolbar",
      "top",
    ],
    "no-return-assign": "error",
    "no-return-await": "error",
    "no-script-url": "error",
    "no-self-assign": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-setter-return": "error",
    "no-shadow-restricted-names": "error",
    "no-sparse-arrays": "warn",
    "no-tabs": 0,
    "no-template-curly-in-string": "warn",
    "no-this-before-super": "error",
    "no-throw-literal": "error",
    "no-undef": "error",
    "no-undef-init": "warn",
    "no-unexpected-multiline": 0,
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": "warn",
    "no-unreachable": "warn",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "warn",
    "no-unused-expressions": Array [
      "error",
      Object {
        "allowShortCircuit": true,
        "allowTaggedTemplates": true,
        "allowTernary": true,
      },
    ],
    "no-unused-vars": Array [
      "warn",
      Object {
        "args": "none",
        "ignoreRestSiblings": true,
      },
    ],
    "no-use-before-define": Array [
      "warn",
      Object {
        "classes": false,
        "functions": false,
        "variables": false,
      },
    ],
    "no-useless-backreference": "warn",
    "no-useless-call": "error",
    "no-useless-computed-key": "warn",
    "no-useless-concat": "error",
    "no-useless-constructor": "warn",
    "no-useless-escape": "warn",
    "no-useless-rename": "warn",
    "no-useless-return": "warn",
    "no-var": "error",
    "no-void": "warn",
    "no-with": "error",
    "object-shorthand": "warn",
    "one-var": Array [
      "warn",
      "never",
    ],
    "operator-assignment": Array [
      "warn",
      "always",
    ],
    "prefer-arrow-callback": "warn",
    "prefer-const": "warn",
    "prefer-destructuring": "warn",
    "prefer-exponentiation-operator": "warn",
    "prefer-numeric-literals": "warn",
    "prefer-object-spread": "warn",
    "prefer-promise-reject-errors": "error",
    "prefer-regex-literals": "error",
    "prefer-rest-params": "error",
    "prefer-spread": "error",
    "prefer-template": "warn",
    "promise/catch-or-return": Array [
      "error",
      Object {
        "allowFinally": true,
        "allowThen": true,
      },
    ],
    "promise/no-nesting": "warn",
    "promise/no-new-statics": "error",
    "promise/no-promise-in-callback": "warn",
    "promise/no-return-in-finally": "warn",
    "promise/no-return-wrap": "error",
    "promise/param-names": "warn",
    "promise/prefer-await-to-callbacks": "warn",
    "promise/prefer-await-to-then": "warn",
    "promise/valid-params": "error",
    "quotes": 0,
    "require-atomic-updates": "warn",
    "require-await": "error",
    "require-unicode-regexp": "error",
    "require-yield": "error",
    "sonarjs/cognitive-complexity": "warn",
    "sonarjs/max-switch-cases": Array [
      "error",
      15,
    ],
    "sonarjs/no-all-duplicated-branches": "error",
    "sonarjs/no-collapsible-if": "error",
    "sonarjs/no-collection-size-mischeck": "warn",
    "sonarjs/no-duplicate-string": "error",
    "sonarjs/no-duplicated-branches": "error",
    "sonarjs/no-element-overwrite": "error",
    "sonarjs/no-extra-arguments": "error",
    "sonarjs/no-identical-conditions": "error",
    "sonarjs/no-identical-expressions": "error",
    "sonarjs/no-identical-functions": "error",
    "sonarjs/no-inverted-boolean-check": "error",
    "sonarjs/no-one-iteration-loop": "error",
    "sonarjs/no-redundant-boolean": "error",
    "sonarjs/no-redundant-jump": "error",
    "sonarjs/no-small-switch": "warn",
    "sonarjs/no-unused-collection": "error",
    "sonarjs/no-use-of-empty-return-value": "error",
    "sonarjs/no-useless-catch": "error",
    "sonarjs/prefer-immediate-return": "error",
    "sonarjs/prefer-object-literal": "error",
    "sonarjs/prefer-single-boolean-return": "error",
    "sonarjs/prefer-while": "error",
    "sort-keys-fix/sort-keys-fix": "warn",
    "spaced-comment": Array [
      "warn",
      "always",
    ],
    "symbol-description": "error",
    "unicorn/better-regex": "error",
    "unicorn/catch-error-name": "error",
    "unicorn/consistent-function-scoping": "error",
    "unicorn/custom-error-definition": "warn",
    "unicorn/error-message": "error",
    "unicorn/escape-case": "error",
    "unicorn/explicit-length-check": "error",
    "unicorn/import-index": "error",
    "unicorn/new-for-builtins": "error",
    "unicorn/no-abusive-eslint-disable": "error",
    "unicorn/no-array-instanceof": "error",
    "unicorn/no-console-spaces": "error",
    "unicorn/no-for-loop": "warn",
    "unicorn/no-hex-escape": "error",
    "unicorn/no-keyword-prefix": "warn",
    "unicorn/no-new-buffer": "error",
    "unicorn/no-object-as-default-parameter": "warn",
    "unicorn/no-process-exit": "error",
    "unicorn/no-unreadable-array-destructuring": "warn",
    "unicorn/no-unsafe-regex": "error",
    "unicorn/no-unused-properties": "warn",
    "unicorn/no-useless-undefined": "error",
    "unicorn/no-zero-fractions": "error",
    "unicorn/prefer-add-event-listener": "error",
    "unicorn/prefer-array-find": "error",
    "unicorn/prefer-dataset": "error",
    "unicorn/prefer-event-key": "error",
    "unicorn/prefer-flat-map": "error",
    "unicorn/prefer-includes": "warn",
    "unicorn/prefer-modern-dom-apis": "error",
    "unicorn/prefer-negative-index": "error",
    "unicorn/prefer-node-append": "error",
    "unicorn/prefer-node-remove": "error",
    "unicorn/prefer-number-properties": "warn",
    "unicorn/prefer-optional-catch-binding": "error",
    "unicorn/prefer-query-selector": "error",
    "unicorn/prefer-reflect-apply": "error",
    "unicorn/prefer-replace-all": "error",
    "unicorn/prefer-set-has": "error",
    "unicorn/prefer-spread": "error",
    "unicorn/prefer-starts-ends-with": "error",
    "unicorn/prefer-string-slice": "error",
    "unicorn/prefer-text-content": "error",
    "unicorn/prefer-trim-start-end": "error",
    "unicorn/prefer-type-error": "error",
    "unicorn/throw-new-error": "error",
    "valid-typeof": "warn",
    "yoda": "warn",
  },
}
`;

exports[`createConfig matches snapshot 1`] = `
Object {
  "env": Object {
    "browser": false,
    "es2020": true,
    "node": true,
  },
  "overrides": Array [
    Object {
      "env": Object {
        "jest": true,
      },
      "extends": Array [
        "plugin:jest-formatting/strict",
      ],
      "files": Array [
        "**/*.?(test|spec).?(ts|js)?(x)",
      ],
      "parserOptions": Object {
        "ecmaVersion": 2020,
      },
      "plugins": Array [
        "jest",
      ],
      "rules": Object {
        "import/no-namespace": "off",
        "jest/consistent-test-it": "off",
        "jest/expect-expect": "off",
        "jest/lowercase-name": "off",
        "jest/no-alias-methods": "off",
        "jest/no-commented-out-tests": "warn",
        "jest/no-conditional-expect": "error",
        "jest/no-deprecated-functions": "error",
        "jest/no-disabled-tests": "warn",
        "jest/no-duplicate-hooks": "warn",
        "jest/no-expect-resolves": "off",
        "jest/no-export": "error",
        "jest/no-focused-tests": "error",
        "jest/no-hooks": "off",
        "jest/no-identical-title": "error",
        "jest/no-if": "error",
        "jest/no-interpolation-in-snapshots": "error",
        "jest/no-jasmine-globals": "error",
        "jest/no-jest-import": "error",
        "jest/no-large-snapshots": Array [
          "warn",
          Object {
            "maxSize": 300,
          },
        ],
        "jest/no-mocks-import": "error",
        "jest/no-restricted-matchers": "off",
        "jest/no-standalone-expect": "off",
        "jest/no-test-callback": "error",
        "jest/no-test-prefixes": "error",
        "jest/no-test-return-statement": "error",
        "jest/no-truthy-falsy": "off",
        "jest/no-try-expect": "off",
        "jest/prefer-called-with": "error",
        "jest/prefer-expect-assertions": "off",
        "jest/prefer-hooks-on-top": "error",
        "jest/prefer-inline-snapshots": "off",
        "jest/prefer-spy-on": "warn",
        "jest/prefer-strict-equal": "warn",
        "jest/prefer-to-be-null": "warn",
        "jest/prefer-to-be-undefined": "warn",
        "jest/prefer-to-contain": "warn",
        "jest/prefer-to-have-length": "warn",
        "jest/prefer-todo": "warn",
        "jest/require-to-throw-message": "off",
        "jest/require-top-level-describe": "error",
        "jest/valid-describe": "error",
        "jest/valid-expect": "error",
        "jest/valid-expect-in-promise": "error",
        "jest/valid-title": "warn",
        "no-empty-function": "off",
      },
      "settings": Object {
        "jest": Object {
          "version": "detect",
        },
      },
    },
  ],
  "parserOptions": Object {
    "sourceType": "module",
  },
  "plugins": Array [
    "import",
    "sort-keys-fix",
    "unicorn",
    "promise",
    "sonarjs",
    "inclusive-language",
  ],
  "rules": Object {
    "array-callback-return": Array [
      "error",
      Object {
        "checkForEach": true,
      },
    ],
    "arrow-body-style": 0,
    "constructor-super": "error",
    "curly": 0,
    "default-case": "error",
    "default-case-last": "error",
    "default-param-last": "error",
    "dot-notation": "warn",
    "eqeqeq": "warn",
    "for-direction": "error",
    "func-names": Array [
      "warn",
      "as-needed",
    ],
    "getter-return": "warn",
    "guard-for-in": "warn",
    "import/default": "warn",
    "import/dynamic-import-chunkname": "warn",
    "import/export": "error",
    "import/first": "error",
    "import/named": "error",
    "import/namespace": "error",
    "import/newline-after-import": "warn",
    "import/no-amd": "error",
    "import/no-anonymous-default-export": "error",
    "import/no-cycle": Array [
      "warn",
      Object {
        "ignoreExternal": true,
        "maxDepth": 5,
      },
    ],
    "import/no-default-export": "warn",
    "import/no-deprecated": "warn",
    "import/no-duplicates": "warn",
    "import/no-dynamic-require": "warn",
    "import/no-extraneous-dependencies": "warn",
    "import/no-mutable-exports": "error",
    "import/no-named-default": "error",
    "import/no-namespace": "error",
    "import/no-self-import": "error",
    "import/no-unresolved": "warn",
    "import/no-unused-modules": "warn",
    "import/no-useless-path-segments": Array [
      "error",
      Object {
        "noUselessIndex": true,
      },
    ],
    "import/no-webpack-loader-syntax": "error",
    "import/order": Array [
      "warn",
      Object {
        "alphabetize": Object {
          "order": "asc",
        },
        "groups": Array [
          Array [
            "builtin",
            "external",
            "internal",
          ],
          Array [
            "unknown",
            "parent",
            "sibling",
          ],
          "index",
        ],
        "newlines-between": "always",
      },
    ],
    "inclusive-language/use-inclusive-words": "error",
    "lines-around-comment": 0,
    "lines-between-class-members": "warn",
    "max-len": 0,
    "new-cap": "warn",
    "no-alert": "error",
    "no-array-constructor": "error",
    "no-async-promise-executor": "error",
    "no-await-in-loop": "error",
    "no-bitwise": "warn",
    "no-caller": "error",
    "no-case-declarations": "warn",
    "no-class-assign": "error",
    "no-compare-neg-zero": "error",
    "no-cond-assign": Array [
      "warn",
      "except-parens",
    ],
    "no-confusing-arrow": 0,
    "no-console": "warn",
    "no-const-assign": "error",
    "no-constant-condition": "error",
    "no-constructor-return": "error",
    "no-control-regex": "warn",
    "no-debugger": "warn",
    "no-div-regex": "warn",
    "no-dupe-args": "error",
    "no-dupe-class-members": "error",
    "no-dupe-else-if": "warn",
    "no-dupe-keys": "warn",
    "no-duplicate-case": "warn",
    "no-else-return": "warn",
    "no-empty": "warn",
    "no-empty-character-class": "warn",
    "no-empty-function": "error",
    "no-empty-pattern": "warn",
    "no-eq-null": "error",
    "no-eval": "error",
    "no-ex-assign": "warn",
    "no-extend-native": "error",
    "no-extra-bind": "warn",
    "no-extra-boolean-cast": "warn",
    "no-fallthrough": "warn",
    "no-func-assign": "warn",
    "no-global-assign": "warn",
    "no-implicit-coercion": Array [
      "warn",
      Object {
        "boolean": false,
        "number": true,
        "string": true,
      },
    ],
    "no-implicit-globals": "error",
    "no-implied-eval": "error",
    "no-import-assign": "error",
    "no-inner-declarations": "warn",
    "no-invalid-regexp": "error",
    "no-invalid-this": "error",
    "no-irregular-whitespace": "warn",
    "no-labels": "error",
    "no-lonely-if": "warn",
    "no-loop-func": "error",
    "no-loss-of-precision": "error",
    "no-misleading-character-class": "warn",
    "no-mixed-operators": 0,
    "no-multi-assign": "error",
    "no-negated-condition": "error",
    "no-new": "error",
    "no-new-func": "error",
    "no-new-object": "error",
    "no-new-wrappers": "error",
    "no-obj-calls": "error",
    "no-octal": "warn",
    "no-octal-escape": "error",
    "no-param-reassign": "error",
    "no-promise-executor-return": "error",
    "no-proto": "error",
    "no-prototype-builtins": "error",
    "no-redeclare": "error",
    "no-regex-spaces": "warn",
    "no-restricted-globals": Array [
      "error",
      "addEventListener",
      "blur",
      "close",
      "closed",
      "confirm",
      "defaultStatus",
      "defaultstatus",
      "event",
      "external",
      "find",
      "focus",
      "frameElement",
      "frames",
      "history",
      "innerHeight",
      "innerWidth",
      "length",
      "location",
      "locationbar",
      "menubar",
      "moveBy",
      "moveTo",
      "name",
      "onblur",
      "onerror",
      "onfocus",
      "onload",
      "onresize",
      "onunload",
      "open",
      "opener",
      "opera",
      "outerHeight",
      "outerWidth",
      "pageXOffset",
      "pageYOffset",
      "parent",
      "print",
      "removeEventListener",
      "resizeBy",
      "resizeTo",
      "screen",
      "screenLeft",
      "screenTop",
      "screenX",
      "screenY",
      "scroll",
      "scrollbars",
      "scrollBy",
      "scrollTo",
      "scrollX",
      "scrollY",
      "self",
      "status",
      "statusbar",
      "stop",
      "toolbar",
      "top",
    ],
    "no-return-assign": "error",
    "no-return-await": "error",
    "no-script-url": "error",
    "no-self-assign": "error",
    "no-self-compare": "error",
    "no-sequences": "error",
    "no-setter-return": "error",
    "no-shadow-restricted-names": "error",
    "no-sparse-arrays": "warn",
    "no-tabs": 0,
    "no-template-curly-in-string": "warn",
    "no-this-before-super": "error",
    "no-throw-literal": "error",
    "no-undef": "error",
    "no-undef-init": "warn",
    "no-unexpected-multiline": 0,
    "no-unmodified-loop-condition": "error",
    "no-unneeded-ternary": "warn",
    "no-unreachable": "warn",
    "no-unsafe-finally": "error",
    "no-unsafe-negation": "warn",
    "no-unused-expressions": Array [
      "error",
      Object {
        "allowShortCircuit": true,
        "allowTaggedTemplates": true,
        "allowTernary": true,
      },
    ],
    "no-unused-vars": Array [
      "warn",
      Object {
        "args": "none",
        "ignoreRestSiblings": true,
      },
    ],
    "no-use-before-define": Array [
      "warn",
      Object {
        "classes": false,
        "functions": false,
        "variables": false,
      },
    ],
    "no-useless-backreference": "warn",
    "no-useless-call": "error",
    "no-useless-computed-key": "warn",
    "no-useless-concat": "error",
    "no-useless-constructor": "warn",
    "no-useless-escape": "warn",
    "no-useless-rename": "warn",
    "no-useless-return": "warn",
    "no-var": "error",
    "no-void": "warn",
    "no-with": "error",
    "object-shorthand": "warn",
    "one-var": Array [
      "warn",
      "never",
    ],
    "operator-assignment": Array [
      "warn",
      "always",
    ],
    "prefer-arrow-callback": "warn",
    "prefer-const": "warn",
    "prefer-destructuring": "warn",
    "prefer-exponentiation-operator": "warn",
    "prefer-numeric-literals": "warn",
    "prefer-object-spread": "warn",
    "prefer-promise-reject-errors": "error",
    "prefer-regex-literals": "error",
    "prefer-rest-params": "error",
    "prefer-spread": "error",
    "prefer-template": "warn",
    "promise/catch-or-return": Array [
      "error",
      Object {
        "allowFinally": true,
        "allowThen": true,
      },
    ],
    "promise/no-nesting": "warn",
    "promise/no-new-statics": "error",
    "promise/no-promise-in-callback": "warn",
    "promise/no-return-in-finally": "warn",
    "promise/no-return-wrap": "error",
    "promise/param-names": "warn",
    "promise/prefer-await-to-callbacks": "warn",
    "promise/prefer-await-to-then": "warn",
    "promise/valid-params": "error",
    "quotes": 0,
    "require-atomic-updates": "warn",
    "require-await": "error",
    "require-unicode-regexp": "error",
    "require-yield": "error",
    "sonarjs/cognitive-complexity": "warn",
    "sonarjs/max-switch-cases": Array [
      "error",
      15,
    ],
    "sonarjs/no-all-duplicated-branches": "error",
    "sonarjs/no-collapsible-if": "error",
    "sonarjs/no-collection-size-mischeck": "warn",
    "sonarjs/no-duplicate-string": "error",
    "sonarjs/no-duplicated-branches": "error",
    "sonarjs/no-element-overwrite": "error",
    "sonarjs/no-extra-arguments": "error",
    "sonarjs/no-identical-conditions": "error",
    "sonarjs/no-identical-expressions": "error",
    "sonarjs/no-identical-functions": "error",
    "sonarjs/no-inverted-boolean-check": "error",
    "sonarjs/no-one-iteration-loop": "error",
    "sonarjs/no-redundant-boolean": "error",
    "sonarjs/no-redundant-jump": "error",
    "sonarjs/no-small-switch": "warn",
    "sonarjs/no-unused-collection": "error",
    "sonarjs/no-use-of-empty-return-value": "error",
    "sonarjs/no-useless-catch": "error",
    "sonarjs/prefer-immediate-return": "error",
    "sonarjs/prefer-object-literal": "error",
    "sonarjs/prefer-single-boolean-return": "error",
    "sonarjs/prefer-while": "error",
    "sort-keys-fix/sort-keys-fix": "warn",
    "spaced-comment": Array [
      "warn",
      "always",
    ],
    "symbol-description": "error",
    "unicorn/better-regex": "error",
    "unicorn/catch-error-name": "error",
    "unicorn/consistent-function-scoping": "error",
    "unicorn/custom-error-definition": "warn",
    "unicorn/error-message": "error",
    "unicorn/escape-case": "error",
    "unicorn/explicit-length-check": "error",
    "unicorn/import-index": "error",
    "unicorn/new-for-builtins": "error",
    "unicorn/no-abusive-eslint-disable": "error",
    "unicorn/no-array-instanceof": "error",
    "unicorn/no-console-spaces": "error",
    "unicorn/no-for-loop": "warn",
    "unicorn/no-hex-escape": "error",
    "unicorn/no-keyword-prefix": "warn",
    "unicorn/no-new-buffer": "error",
    "unicorn/no-object-as-default-parameter": "warn",
    "unicorn/no-process-exit": "error",
    "unicorn/no-unreadable-array-destructuring": "warn",
    "unicorn/no-unsafe-regex": "error",
    "unicorn/no-unused-properties": "warn",
    "unicorn/no-useless-undefined": "error",
    "unicorn/no-zero-fractions": "error",
    "unicorn/prefer-add-event-listener": "error",
    "unicorn/prefer-array-find": "error",
    "unicorn/prefer-dataset": "error",
    "unicorn/prefer-event-key": "error",
    "unicorn/prefer-flat-map": "error",
    "unicorn/prefer-includes": "warn",
    "unicorn/prefer-modern-dom-apis": "error",
    "unicorn/prefer-negative-index": "error",
    "unicorn/prefer-node-append": "error",
    "unicorn/prefer-node-remove": "error",
    "unicorn/prefer-number-properties": "warn",
    "unicorn/prefer-optional-catch-binding": "error",
    "unicorn/prefer-query-selector": "error",
    "unicorn/prefer-reflect-apply": "error",
    "unicorn/prefer-replace-all": "error",
    "unicorn/prefer-set-has": "error",
    "unicorn/prefer-spread": "error",
    "unicorn/prefer-starts-ends-with": "error",
    "unicorn/prefer-string-slice": "error",
    "unicorn/prefer-text-content": "error",
    "unicorn/prefer-trim-start-end": "error",
    "unicorn/prefer-type-error": "error",
    "unicorn/throw-new-error": "error",
    "valid-typeof": "warn",
    "yoda": "warn",
  },
}
`;

exports[`getDependencies accepts alternative tsdefaultConfig path 1`] = `
Object {
  "hasJest": true,
  "hasJestDom": false,
  "hasNodeTypes": false,
  "hasTestingLibrary": false,
  "react": Object {
    "hasReact": false,
    "isNext": false,
    "isPreact": false,
    "version": undefined,
  },
  "typescript": Object {
    "hasTypeScript": true,
    "version": "1.0.0",
  },
}
`;

exports[`getDependencies matches snapshot 1`] = `
Object {
  "hasJest": true,
  "hasJestDom": false,
  "hasNodeTypes": false,
  "hasTestingLibrary": false,
  "react": Object {
    "hasReact": false,
    "isNext": false,
    "isPreact": false,
    "version": undefined,
  },
  "typescript": Object {
    "hasTypeScript": false,
    "version": "4.0.2",
  },
}
`;

exports[`getDependencies matches snapshot when erroring 1`] = `
Object {
  "hasJest": false,
  "hasJestDom": false,
  "hasNodeTypes": false,
  "hasTestingLibrary": false,
  "react": Object {
    "hasReact": false,
    "isNext": false,
    "isPreact": false,
    "version": undefined,
  },
  "typescript": Object {
    "hasTypeScript": false,
    "version": undefined,
  },
}
`;
